src/modules/users/follow.service.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { Errors } from 'src/common/errors';
+
+import { FollowDto } from './dto/follow.dto';
+import { User } from './user.entity';
+
+@Injectable()
+export class FollowService {
+  constructor(
+    @InjectRepository(User)
+    private userRepository: Repository<User>,
+  ) {}
+
+  async followVendor(followDto: FollowDto): Promise<User> {
+    try {
+      const buyer = await this.userRepository.findOne({
+        where: { id: followDto.buyerId },
+        relations: ['following'],
+      });
+
+      const vendor = await this.userRepository.findOne({
+        where: { id: followDto.vendorId },
+      });
```

Comment:
you can do it in one query with queary builder.

Reply:
changed here https://github.com/ZenBit-Tech/code-lions_be/pull/35/commits/38bf9151bd5a4ca9bb7615b4d7a8ab8c1769033e